### ISSUE TITLE
[FG:InPlacePodVerticalScaling] pull-kubernetes-e2e-capz-windows-master test fail with InPlacePodVerticalScaling Beta

### ISSUE THREAD

**Issue by esotsal** [kubernetes/kubernetes#128897]
### Which jobs are failing?

pull-kubernetes-e2e-capz-windows-master 

https://prow.k8s.io/view/gs/kubernetes-ci-logs/pr-logs/pull/128880/pull-kubernetes-e2e-capz-windows-master/1859240005449289728

### Which tests are failing?

Kubernetes e2e suite: [It] [sig-storage] 
- Projected secret should be consumable from pods in volume [NodeConformance] [Conformance] 
-Secrets should be consumable from pods in volume [NodeConformance] [Conformance] 
-ConfigMap should be consumable from pods in volume with mappings [NodeConformance] [Conformance] expand_more 	23s
-Secrets should be consumable from pods in volume with mappings [NodeConformance] [Conformance] expand_more 	33s
-Projected configMap should be consumable from pods in volume with mappings [NodeConformance] [Conformance] expand_more 	25s
-  Projected secret should be able to mount in a volume regardless of a different secret existing with same name in different namespace [NodeConformance] expand_more 	25s
- Projected configMap should be consumable from pods in volume as non-root [NodeConformance] [Conformance] expand_more 	27s
- Projected configMap should be consumable from pods in volume [NodeConformance] [Conformance] expand_more 	31s
- Projected secret should be consumable in multiple volumes in a pod [NodeConformance] [Conformance] expand_more 	27s
- Projected secret should be consumable from pods in volume with mappings [NodeConformance] [Conformance] expand_more 	27s
- Secrets should be able to mount in a volume regardless of a different secret existing with same name in different namespace [NodeConformance] [Conformance] expand_more 	25s
- Secrets should be consumable in multiple volumes in a pod [NodeConformance] [Conformance] expand_more 	27s
- ConfigMap should be consumable from pods in volume [NodeConformance] [Conformance] expand_more 	1m9s
- ConfigMap should be consumable from pods in volume as non-root [NodeConformance] [Conformance] expand_more 	25s
- ConfigMap should be consumable from pods in volume with mappings as non-root [NodeConformance] [Conformance] expand_more 	27s
- Projected configMap should be consumable from pods in volume with mappings as non-root [NodeConformance] [Conformance] expand_more

Kubernetes e2e suite: [It] [sig-windows] [Feature:Windows]
- Windows volume mounts check volume mount permissions container should have readOnly permissions on emptyDir expand_more 	35s

Kubernetes e2e suite: [It] [sig-node]
- Container Runtime blackbox test on terminated container should report termination message if TerminationMessagePath is set as non-root user and at a non-default path [NodeConformance] [Conformance] expand_more 	10m3s

### Since when has it been failing?

Last successful 12 November 20:20 1856416666905219072 
https://prow.k8s.io/view/gs/kubernetes-ci-logs/logs/ci-kubernetes-e2e-capz-master-windows/1856416666905219072

First failed 12 November 23:20 1856461965182898176 
https://prow.k8s.io/view/gs/kubernetes-ci-logs/logs/ci-kubernetes-e2e-capz-master-windows/1856461965182898176

### Testgrid link

https://testgrid.k8s.io/sig-release-master-informing#capz-windows-master

### Relevant SIG(s)

/sig-node /sig-storage /sig-windows

**Comment by dims** [comment]
`capz-windows-master` has recovered https://testgrid.k8s.io/sig-release-master-informing#capz-windows-master&width=20

**Comment by kannon92** [comment]
I think this ticket is for investigating why inplace broke this job.

/triage accepted
/priority important-soon

This was the main reason why I was +1 on the revert as I think we had a decent understanding of the test problem but this one is perplexing to me.

**Comment by esotsal** [comment]
> This was the main reason why I was +1 on the revert as I think we had a decent understanding of the test problem but this one is perplexing to me.

Correct , the kubetest2 failure discussed [here](https://github.com/kubernetes/kubernetes/pull/128880#issuecomment-2490066484) is also the other reason . We need to solve both those problems b put back InPlacePodVerticalScaling beta . I strongly think both those relate with test code . #127951 

/cc @tallclair @AnishShah

**Comment by esotsal** [comment]
> `capz-windows-master` has recovered https://testgrid.k8s.io/sig-release-master-informing#capz-windows-master&width=20

This run from https://prow.k8s.io/pr-history/?org=kubernetes&repo=kubernetes&pr=128880 passes without the revert https://prow.k8s.io/view/gs/kubernetes-ci-logs/pr-logs/pull/128880/pull-kubernetes-e2e-capz-windows-master/1859388054804893696

**Comment by carlory** [comment]
I do some tests on Windows via https://github.com/kubernetes/kubernetes/pull/128876 and https://github.com/kubernetes/kubernetes/pull/128927. 

Create a pod with a configmap volume and check the permission of the volume. 

It shows that the permission of the volume is correct. the expected permission is `rwxrwxr-x` and the actual permission is `rwxrwxr-x`.

```
Path   : Microsoft.PowerShell.Core\FileSystem::C:\etc\configmap-volume\path\to\
         data-2
Owner  : BUILTIN\Administrators
Group  : NT AUTHORITY\SYSTEM
Access : NT AUTHORITY\Authenticated Users Allow  Modify, Synchronize
         NT AUTHORITY\SYSTEM Allow  FullControl
         BUILTIN\Administrators Allow  FullControl
         BUILTIN\Users Allow  ReadAndExecute, Synchronize
Audit  : 
Sddl   : O:BAG:SYD:AI(A;ID;0x1301bf;;;AU)(A;ID;FA;;;SY)(A;ID;FA;;;BA)(A;ID;0x12
         00a9;;;BU)
```

It shows that the permission of the volume is incorrect. the expected permission is `rwxrwxr-x` but the actual permission is `rwx---r-x`.

```
Path   : Microsoft.PowerShell.Core\FileSystem::C:\etc\configmap-volume\path\to\
         data-2
Owner  : BUILTIN\Administrators
Group  : NT AUTHORITY\SYSTEM
Access : BUILTIN\Administrators Allow  FullControl
         S-1-5-21-3344892312-2374761293-1590377129-513 Allow  ReadAndExecute, 
         Synchronize
         BUILTIN\Users Allow  ReadAndExecute, Synchronize
Audit  : 
Sddl   : O:BAG:SYD:AI(A;ID;FA;;;BA)(A;ID;0x1200a9;;;S-1-5-21-3344892312-2374761
         293-1590377129-513)(A;ID;0x1200a9;;;BU)
```

We can see the difference on access control list. I didn't know the reason why other tests affected the directory permission. https://github.com/kubernetes/kubernetes/pull/128880 may not fix the root cause of the issue.

**Comment by esotsal** [comment]
Thanks @carlory , comparing runs from your tests, i am a bit puzzled now.  Checking [PR runs without IPPVS](https://prow.k8s.io/pr-history/?org=kubernetes&repo=kubernetes&pr=128876)  [PR runs with IPPVS ](https://prow.k8s.io/pr-history/?org=kubernetes&repo=kubernetes&pr=128927) from your test commits. I notice you don't have a successful run with IPPVs , [but in #12880 we had some green runs](https://prow.k8s.io/pr-history/?org=kubernetes&repo=kubernetes&pr=128880) ( so it might not fix the root cause , but could it be that reduces the flakiness? ) . Does this mean that InPlacePodVerticalScaling is not the root cause but InPlacePodVerticalScaling increases the number of flake tests ?  .

**Comment by carlory** [comment]
> does this mean that it might not fix the root cause , but could it be that reduces the flakiness?.

@esotsal  Yes, it is. #128880 only changes the test code, not the feature itself. Users may encounter the same issue when their clusters enable IPPVs.

To be honest, I'm very confused about why #128880 reduces the flakiness or resolves this issue, but the fact is that it does because the ci is green again.

The `Access` list I mentioned in the previous comment may help some developers understand what happened in the test.

**Comment by hshiina** [comment]
Though I'm not sure this is related to the failures, there are the following messages in [kubelet.log](https://storage.googleapis.com/kubernetes-ci-logs/pr-logs/pull/128880/pull-kubernetes-e2e-capz-windows-master/1859240005449289728/artifacts/clusters/capz-conf-k7qm5f/machines/capz-conf-k7qm5f-md-win-c7zsz-927rb/kubelet.log) in failed cases:
```
E1120 14:36:37.747170    2972 kuberuntime_manager.go:681] "Unable to get resource configuration" pod="cloud-node-manager-windows-n9klf"
E1120 14:36:37.747170    2972 pod_workers.go:1301] "Error syncing pod, skipping" err="failed to SyncPod: unable to get resource configuration processing resize for pod cloud-node-manager-windows-n9klf" pod="kube-system/cloud-node-manager-windows-n9klf" podUID="fe6bd320-6e2e-481f-9377-c55cb58a4907"
```

It looks that `doPodResizeAction()` was called unexpectedly on windows:
https://github.com/kubernetes/kubernetes/blob/35d098aaa0a912171c46345e2c0a7b54d6046ef4/pkg/kubelet/kuberuntime/kuberuntime_manager.go#L676-L684
There may be something wrong with this feature regarding Windows.

**Comment by esotsal** [comment]
[/priotity ](label:priority/critical-urgent)  even if InPlacePodVerticalScaling is not Beta , if end user enables InPlacePodVerticalScaling it will have the same consequences on a Windows system.

**Comment by kannon92** [comment]
> Should we increase priority to critical-urgent @pacoxu ?

Given that the feature is alpha, we don't cherry-pick bugs for alpha features so I don't see the need at that moment.

/priority important-soon

This means we want this for 1.33 which I think is what we want.

**Comment by esotsal** [comment]
> > Should we increase priority to critical-urgent @pacoxu ?
> 
> Given that the feature is alpha, we don't cherry-pick bugs for alpha features so I don't see the need at that moment.
> 
> /priority important-soon
> 
> This means we want this for 1.33 which I think is what we want.

Thanks for clarifying!

**Comment by pacoxu** [comment]
Is this possible to be related to https://github.com/kubernetes/kubernetes/pull/129083?

- I try to do some test based on current master.

**Comment by pacoxu** [comment]
> Is this possible to be related to #129083?
> 
> * I try to do some test based on current master.

Confirmed. The issue is not related.  Current master + enabling VPA beta will fail for https://prow.k8s.io/view/gs/kubernetes-ci-logs/pr-logs/pull/128040/pull-kubernetes-e2e-capz-windows-master/1867106924625924096.

**Comment by tallclair** [comment]
I think https://github.com/kubernetes/kubernetes/pull/129214 should address this.

**Comment by esotsal** [comment]
@tallclair , we reopened it during todays SIG Node CI due to https://github.com/kubernetes/kubernetes/pull/129242, should we assign this issue to you ?

**Comment by tallclair** [comment]
Yes. https://github.com/kubernetes/kubernetes/pull/129242#issuecomment-2583399303 should address this.
